pkg/sql/pgwire/hba/conf.rl

```diff
@@ -144,7 +147,7 @@ func Parse(input string) (*Conf, error) {
 			conf.Entries = append(conf.Entries, e)
 		}
 		host =
-			'host' %newHost ws
+			('host' %newHost | 'local' %newLocal) ws
```

Comment:
The `local` connection type, as defined by the postgres docs, only has 3 following fields as opposed to host which has 4 or 5. I would prefer not adding local until our parser correctly supports the expected number of fields that local needs.

I'm reaching the point now where I no longer think ragel is a good fit for this problem. It was an experiment, and I consider it failed. It takes a long time to figure out how to use the language, and even then it is very easy to make blind mistakes. When we need to extend it 6-12 months later (as in here), it is difficult for the reviewer (me in this case, even though I was the original author) to remember how ragel works. The change you made a few days ago that correctly parsed single-letter comma-separated things for example, I didn't really understand what was going on, but it passed the tests so I hit accept.

Unless you really strongly disagree, or have a clean way to add the correct parsing for `local`, I suggest removing this addition.

https://blog.cloudflare.com/incident-report-on-memory-leak-caused-by-cloudflare-parser-bug/ seems like cloudflare had similar problems with ragel wrt maintainability.

Reply:
Oh, I meant to say that I'm very on board with rewriting this parser in Go. It's not a super complicated language. It's annoying since it's sometimes space-separated except with double quotes. But I believe we could look at the pg implementation for some quick guidance about a well-tested way to write our parser.